src/components/ChatEntry.js

```diff
@@ -1,22 +1,43 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp'
 
 const ChatEntry = (props) => {
+
+
+  const updateLikeButtonClick = () =>{
+    const updateChat ={
+      id : props.id,
+      sender : props.sender,
+      body : props.body,
+      liked : !props.liked, 
+      timeStamp : props.timeStamp,
+    }
+    props.onUpdate(updateChat);
```

Comment:
👀 This style of creating the new object to go in our list is how Learn showed working with the student attendance example. However, it turns out we don't need to do this when we send the id of the record we want to report has changed. The `id` passed there can come directly from props, as
```js
    props.onUpdate(props.id);
```

Our local component here shouldn't need to know how to toggle the like status. Rather, this component should pass the id of the message whose like button was clicked to the callback, and _that_ logic should be responsible for determining what the new liked status for the message should be. This would help ensure we have a single source of truth (the list of messages), from which we can draw the entire UI.
```js
  // in App.js
  const handleLikeClicked = (id) => {
    // copy the list of message stored in the App state
    // copy the message indicated by the id
    // flip the liked status for the message
    // update the App state tracking the messages
  };

  // later in App.js
  <ChatLog entries={messages} onLikeClicked={handleLikeClicked} />